src/Config/Config.php

```diff
@@ -8,779 +8,395 @@
-         */
-        public function __unset($key): void
-        {
-            $this->remove($key);
-        }
-
-        /**
-         * @param string|int $key
-         *
-         * @return bool
-         */
-        public function has($key): bool
-        {
-            return isset($this->data[$key]);
-        }
-
-        /**
-         * @param string|int $key
-         *
-         * @return bool
-         */
-        public function __isset($key): bool
-        {
-            return $this->has($key);
-        }
+    /**
+     * @return void
+     */
+    public function __clone()
+    {
+        $data = [];
 
-        /**
-         * @return array
-         */
-        public function toArray(): array
-        {
-            $array = [];
-            $data = $this->data;
-
-            foreach ($data as $key => $value) {
-                if ($value instanceof self) {
-                    $value = $value->toArray();
-                }
-                $array[$key] = $value;
+        foreach ($this->data as $key => $value) {
+            if ($value instanceof self) {
+                $value = clone $value;
             }
-
-            return $array;
+            $data[$key] = $value;
         }
 
-        /**
-         * @inheritDoc
-         */
-        public function current(): mixed
-        {
-            return current($this->data);
-        }
+        $this->data = $data;
+    }
 
+    /**
+     * @param mixed $key
+     * @param mixed $default
+     * @return mixed|null
+     */
+    public function get(mixed $key, mixed $default = null)
+    {
+        return $this->data[$key] ?? $default;
+    }
 
-        /**
-         * @inheritDoc
-         *
-         * @return void
-         */
-        public function next(): void
-        {
-            next($this->data);
-        }
+    /**
+     * @param string|integer $key
+     * @return mixed|null
+     */
+    public function __get($key)  // phpcs:ignore
+    {
+        return $this->get($key);
+    }
 
-        /**
-         * @inheritDoc
-         *
-         * @return
-         */
-        public function key(): mixed
-        {
-            return key($this->data);
+    /**
+     * @param mixed $key
+     * @param mixed $value
+     * @return void
+     */
+    public function set(mixed $key, mixed $value): void
+    {
+        if (true === is_array($value)) {
+            $value = new self($value);
         }
 
-        /**
-         * @inheritDoc
-         *
-         * @return bool
-         */
-        public function valid(): bool
-        {
-            return null !== $this->key();
-        }
+        $this->data[$key] = $value;
+    }
 
-        /**
-         * @inheritDoc
-         *
-         * @return void
-         */
-        public function rewind(): void
-        {
-            reset($this->data);
-        }
+    /**
+     * @param mixed $key
+     * @param mixed $value
+     * @return void
+     */
+    public function __set($key, $value): void // phpcs:ignore
```

Comment:
Should this be `public function __set(mixed $key, mixed $value): void`? I assume the phpcs:ignore was added because of the missing type hints. Since we're targeting PHP 8.1+, adding mixed should be fine and we can likely remove the ignore.